### PR TITLE
Support Python 3.9, drop Python 3.6

### PR DESCRIPTION
**Pull request by MarcoGorelli**

### CODE REVIEW

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
LGTM, just needs a line in the release-notes.

**Comment by MarcoGorelli** [comment]
Suggestions on how to phrase it?
```
## PyMC3 4.0.0 (on deck)

This is the first release to support Python3.9 and to drop Python3.6.

### Maintenance
- Fixed bug whereby partial traces returns after keyboard interrupt during parallel sampling had fewer draws than would've been available [#4318](https://github.com/pymc-devs/pymc3/pull/4318)
- Make `sample_shape` same across all contexts in `draw_values` (see [#4305](https://github.com/pymc-devs/pymc3/pull/4305)).
- Removed `theanof.set_theano_config` because it illegally touched Theano's privates (see [#4329](https://github.com/pymc-devs/pymc3/pull/4329)).
```

**Comment by twiecki** [comment]
I think we can put this into `3.11`. CC @michaelosthege

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Release Notes
 
-## PyMC3 3.10.1 (on deck)
+## PyMC3 4.0.0 (on deck)
+
+This is the first release to support Python3.9 and to drop Python3.6.
```
Comment (michaelosthege) [inline RELEASE-NOTES.md:5]: I will change this in #4214 to become 3.11.

**Review by michaelosthege — APPROVED** [review]

**Pull request merged**